src/middleware/authProvider/index.js

```diff
@@ -0,0 +1,116 @@
+import url from 'url';
+import passport from 'passport';
+import { Strategy, Issuer } from 'openid-client';
+import session from 'express-session';
+import redis from 'redis';
+import connectRedis from 'connect-redis';
+
+export class AuthProvider {
```

Comment:
do you think we should add logging in here? or in the health routes? might be useful debug

Reply:
We do have logger in the healthRoutes. what kind of logging do you think will be valuable here? Just `logger.info({{ name of the function called }})`?

Reply:
sorry I meant the auth routes

Reply:
yeah just something in the login/logout functions so we can follow the calls through rather than rely on isapi logs 🤷

Reply:
ok I'll add something into the authRoutes